test/onnx/test_pytorch_onnx_onnxruntime.py

```diff
@@ -9503,27 +9503,35 @@ def make_input(batch_size):
             inputs = [torch.randn(l, RNN_INPUT_SIZE) for l in seq_lengths]
             inputs = rnn_utils.pad_sequence(inputs, batch_first=batch_first)
             inputs = [inputs]
-
+            input_names = ["input"]
             directions = 2 if bidirectional else 1
 
             if initial_state:
                 h0 = torch.randn(directions * layers, batch_size, RNN_HIDDEN_SIZE)
                 c0 = torch.randn(directions * layers, batch_size, RNN_HIDDEN_SIZE)
                 inputs.append((h0, c0))
+                input_names.append("h0")
+                input_names.append("c0")
             if packed_sequence != 0:
-                inputs.append(torch.IntTensor(seq_lengths))
+                inputs.append(torch.LongTensor(seq_lengths))
+                input_names.append("seq_lengths")
             if len(inputs) == 1:
                 input = inputs[0]
             else:
                 input = tuple(inputs)
-            return input
+            return input, input_names
 
-        input = make_input(RNN_BATCH_SIZE)
-        self.run_test(model, input)
+        input, input_names = make_input(RNN_BATCH_SIZE)
+        dynamic_axes = {"input": [0, 1], "seq_lengths": [0]}
+        if initial_state:
+            dynamic_axes.update({"h0": [1], "c0": [1]})
+        export_options = {"input_names": input_names, "dynamic_axes": dynamic_axes}
 
         # test that the model still runs with a different batch size
-        other_input = make_input(RNN_BATCH_SIZE + 1)
-        self.run_test(model, other_input)
+        other_input, _ = make_input(RNN_BATCH_SIZE + 1)
+        self.run_test(
+            model, input, additional_test_inputs=[other_input], **export_options
```

Comment:
So basically it's adding dynamic axes tests into RNN. Have you tried the new tests with the other two? elman/gru in test cases? cc @BowenBao

Reply:
Yes I did  - and I know for sure the CI ran them too, as they were failing at some point.

Reply:
Not just adding axes, but actually running ONNX exported with one size, with input of different size(well, at least the batch size), to make sure it's not exported as constant as it was the case before because of as_tensor().